recipe-server/client/control/state/app/session/actions.js

```diff
@@ -61,13 +61,17 @@ export function saveSession() {
   };
 }
 
-export function addSessionView(category, caption) {
+export function addSessionView(category, caption, identicon) {
   return async (dispatch, getState) => {
-    const url = getState().router.pathname;
+    let url = getState().router.pathname;
+
+    // Prevent exact subpages (e.g. edit, clone pages) from appearing in the nav.
+    // Instead, this will link to the 'view' page for that recipe/revision.
+    url = url.replace(/\/(edit|clone|approval_history)/, '');
```

Comment:
This seems fragile. It requires this function to know about all the rest of the code, and makes it less re-usable. I think this should be refactored so that callers pass the URL. That way each individual caller can manager how it's URLs need to be deduplicated.

Either that, or this sort of "parent page" information should be stored on the route, so it is stored in a central place with the routing information.

Reply:
I've gone with a bit of a combination of both options, seen in the latest commit: https://github.com/mozilla/normandy/pull/1076/commits/514f3359fbc23074975e6f28ed61baa1d2924604 

On the route definitions, I use an `ignoreSession` variable to determine if it should handle changing the reported URL. Then, in order to know what the 'correct' URL is, I put together a `getNamedRoute` function which traverses the route tree and finds the specified path based on the given crumb. For example:

```
getNamedRoute('New Extension') === '/extension/new/'
getNamedRoute('View Recipe') === '/recipe/:recipeId/'
getNamedRoute('View Recipe', { recipeId: 3 }) === '/recipe/3/'
```

This `getNamedRoute` function will actually be useful in a few places, so I quite like this approach. I still have to write tests, but this code/commit is certainly open for feedback.